radix-engine/src/model/resources/executables/resource_manager.rs

```diff
@@ -276,8 +281,25 @@ fn build_substates(
         access_rules_map
             .remove(&UpdateNonFungibleData)
             .unwrap_or((AllowAll, rule!(deny_all)));
+    let (update_metadata_access_rule, update_metadata_mutability) = access_rules_map
+        .remove(&UpdateMetadata)
+        .unwrap_or((DenyAll, rule!(deny_all)));
 
     let mut access_rules = AccessRules::new();
+    access_rules.set_access_rule_and_mutability(
+        AccessRuleKey::Native(NativeFn::Method(NativeMethod::Metadata(
+            MetadataMethod::Set,
+        ))),
+        update_metadata_access_rule,
+        update_metadata_mutability,
+    );
+    access_rules.set_access_rule_and_mutability(
+        AccessRuleKey::Native(NativeFn::Method(NativeMethod::Metadata(
+            MetadataMethod::Get,
+        ))),
+        AllowAll,
+        DenyAll,
```

Comment:
`AllowAll` + `DenyAll` here is by design, even for the `WithOwner` variant, right?

Reply:
Yup, that's right.